cobalt/ifa/ifa.cc

```diff
@@ -0,0 +1,99 @@
+// Copyright 2025 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "cobalt/ifa/ifa.h"
+
+#include <string>
+
+#include "base/logging.h"
+#include "base/memory/singleton.h"
+#include "base/notreached.h"
+#include "starboard/common/system_property.h"
+#include "starboard/system.h"
+
+namespace cobalt {
+namespace ifa {
+
+// static
+Ifa* Ifa::GetInstance() {
+  return base::Singleton<Ifa, base::LeakySingletonTraits<Ifa>>::get();
+}
+
+Ifa::Ifa() {
+  ifa_extension_ = static_cast<const StarboardExtensionIfaApi*>(
+      SbSystemGetExtension(kStarboardExtensionIfaName));
+  if (!ifa_extension_) {
+    return;
+  }
+  DCHECK_EQ(std::string(ifa_extension_->name), kStarboardExtensionIfaName)
+      << "Unexpected extension name.";
+  DCHECK_GE(ifa_extension_->version, 1) << "Unexpected extension version.";
+
+  if (ifa_extension_->version >= 2) {
```

Comment:
@yell0wd0g I need to find a good way to mimic or replace the behavior in c25, see https://source.corp.google.com/h/github/youtube/cobalt/+/25.lts.1+:cobalt/h5vcc/h5vcc_system.cc;l=45-53?q=h5vcc_system.cc&ss=h%2Fgithub%2Fyoutube%2Fcobalt%2F%2B%2Frefs%2Fheads%2F25.lts.1%2B where we're tracking promises and resolving them all at once like so. Haven't investigated much here but could probably use your help on it next week.

Reply:
Believe I have the behavior here